paper-server/src/main/java/org/bukkit/craftbukkit/map/CraftMapCanvas.java

```diff
@@ -92,6 +92,9 @@ protected byte[] getBuffer() {
     @Override
     public void drawImage(int x, int y, Image image) {
         // Paper start - Reduce work done by limiting size of image and using System.arraycopy
+        if (x < 0 || y < 0 || x >= 128 || y >= 128)
```

Comment:
not sure about this, did this logic allow using negative values to support offsetting an oversized values?

Reply:
I didn't think about that, I added that because the current impl will throw if a negative value is supplied. does seem like the old impl works with negative values, so I'll try seeing if I can fix that too

Reply:
Tested this with a bunch of offsets & image sizes and it works the same as it did in CB now